linkerd/proxy/transport/src/listen.rs

```diff
@@ -26,6 +26,12 @@ pub trait Bind<T> {
     type Incoming: Stream<Item = Result<(Self::Addrs, Self::Io)>> + Send + Sync + 'static;
 
     fn bind(self, params: &T) -> Result<Bound<Self::Incoming>>;
+    fn bind_additional(self, _params: &T) -> Result<Bound<Self::Incoming>>
+    where
+        Self: Sized,
+    {
+        unimplemented!();
+    }
```

Comment:
We should not leave this in a way that can lead to runtime panics. If we really do need to modify the interface, we should force all implementations to handle it. But I don't think it's necessary to modify this interface.

I recall that before--I can't find where, maybe it was verbal--I had suggested that we implement all of this _behind_ the single bind() call.  I.e. we should need minimal application changes to support this and this change can be much smaller and less invasive.